picard/util/preservedtags.py

```diff
@@ -40,11 +40,11 @@ def _normalize_tag(tag):
         return tag.strip().lower()
 
     def add(self, name):
-        self._tags.add(name)
+        self._tags.add(self._normalize_tag(name))
         self._to_config()
 
     def discard(self, name):
-        self._tags.discard(name)
+        self._tags.discard(self._normalize_tag(name))
         self._to_config()
 
     def __contains__(self, key):
```

Comment:
Imho, we should use `self._normalize_tag(key)` instead of `key.lower()` here too

Reply:
Agreed. I changed this and added a test check for this